main.cpp

```diff
@@ -850,17 +849,18 @@ class Framework {
       }
 
       if (print_memory_usage) {
-        lv_mem_monitor(&mem_mon);
-        if (mem_mon.free_size != mem_mon_last_free_size) {
+        auto currentFreeHeap = xPortGetFreeHeapSize();
+        auto minimumEverFreeHeap = xPortGetMinimumEverFreeHeapSize();
+        if (currentFreeHeap != lastFreeHeapSize) {
```

Comment:
we can move the minimum value getter into the if block to only get that value when we really need it

```suggestion
        if (currentFreeHeap != lastFreeHeapSize) {
          auto minimumEverFreeHeap = xPortGetMinimumEverFreeHeapSize();
```